### PR TITLE
added cross derivatives for CrossTermBarrierFunctionMaterial #5192

### PR DESCRIPTION
**Pull request by jbair34**
Added the necessary changes to incorporate cross term derivatives for the second order derivatives.

### CODE REVIEW

**Comment by jwpeterson** [comment]
I'd like to squash these two commits together before merge.

**Comment by jwpeterson** [comment]
I'll take care of it as soon as @dschwen signs off.

File: modules/phase_field/src/materials/CrossTermBarrierFunctionMaterial.C [hunk 1]
```diff
@@ -74,10 +86,13 @@ CrossTermBarrierFunctionMaterial::computeQpProperties()
...
       {
         case 0: // SIMPLE
-          _prop_g[_qp]         +=  _W_ij[_num_eta*i + j] * (*_eta[i])[_qp] * (*_eta[i])[_qp] * (*_eta[j])[_qp] * (*_eta[j])[_qp];
-          (*_prop_dg[i])[_qp]  +=  2 * _W_ij[_num_eta*i + j] * 2 * (*_eta[i])[_qp] * (*_eta[j])[_qp] * (*_eta[j])[_qp];
-          (*_prop_d2g[i])[_qp] +=  2 * _W_ij[_num_eta*i + j] * 2 * (*_eta[j])[_qp] * (*_eta[j])[_qp];
-          break;
+          _prop_g[_qp]         +=  _W_ij[_num_eta * i + j] * (*_eta[i])[_qp] * (*_eta[i])[_qp] * (*_eta[j])[_qp] * (*_eta[j])[_qp];
+          (*_prop_dg[i])[_qp]  +=  _W_ij[_num_eta * i + j] * 2 * (*_eta[i])[_qp] * (*_eta[j])[_qp] * (*_eta[j])[_qp];
+          if (i == j)
+            (*_prop_d2g[i][j])[_qp] +=  _W_ij[_num_eta * i + j] * 2 * (*_eta[j])[_qp] * (*_eta[j])[_qp];
```
Comment (dschwen) [inline modules/phase_field/src/materials/CrossTermBarrierFunctionMaterial.C]: I'm not sure this is correct. Both nested loops go from 0 to _num_eta and you acces ```_prop_d2g[i][j]. **However*, please note that ```_prop_d2g[i][j]``` and  ```_prop_d2g[j][i]``` point to the exact same memory location. So you are probably overcounbting here.

Comment (jbair34) [inline modules/phase_field/src/materials/CrossTermBarrierFunctionMaterial.C]: So these two _prop_d2g terms should be divided by two to account for the [i][j] and [j][i] contributions?

Comment (dschwen) [inline modules/phase_field/src/materials/CrossTermBarrierFunctionMaterial.C]: Or have ```j``` iterate from ```i``` to ```_num_eta```

**Comment by permcody** [comment]
@friedmud or @brianmoose - Do you know what happened here? This PR finished testing but it shows in progress still.

**Comment by jwpeterson** [comment]
Squashed and merged to devel in 5532a58.

**Pull request closed without merging**